### PR TITLE
[metricbeat] migrate kafka to reporterV2 with error return

### PR DESCRIPTION
**Pull request by fearful-symmetry**
see elastic/beats#11374 

For whatever reason, I couldn't get the integration tests working on OSX. Took me a while to get a proper beats env setup on my linux box.

### CODE REVIEW

File: metricbeat/module/kafka/partition/partition.go [hunk 1]
```diff
@@ -163,16 +164,20 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
...
-				r.Event(mb.Event{
+				sent := r.Event(mb.Event{
 					ModuleFields: common.MapStr{
 						"broker": evtBroker,
 						"topic":  evtTopic,
 					},
 					MetricSetFields: event,
 				})
+				if !sent {
+					return errors.New("metricset is closed")
```
Comment (exekias) [inline metricbeat/module/kafka/partition/partition.go]: Should we report an error here? I understood this is `false` when the metricset is stopped, which is not an error but a known condition

Comment (ruflin) [reply metricbeat/module/kafka/partition/partition.go]: @exekias I think you are correct: https://github.com/elastic/beats/issues/11666

Comment (fearful-symmetry) [reply metricbeat/module/kafka/partition/partition.go]: Yah. In theory you _don't_ need to return an error, but for whatever reason `return nil` just seemed...weird here.

Comment (jsoriano) [reply metricbeat/module/kafka/partition/partition.go]: We should `return nil` here, yes. If you want you can log something like "Fetch interrupted, metricset stopping" at the debug level.

**Pull request merged**